sources/api/early-boot-config/src/provider/local_file.rs

```diff
@@ -0,0 +1,66 @@
+//! The local_file module implements the `PlatformDataProvider` trait for gathering userdata from
+//! local file
+
+use super::{PlatformDataProvider, SettingsJson};
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+
+pub(crate) struct LocalFileDataProvider;
+
+impl LocalFileDataProvider {
+    pub(crate) const USER_DATA_FILE: &'static str = "/etc/early-boot-config/user-data";
```

Comment:
Does this need to be `pub(crate)`?

Reply:
I believe so - we use the existence of this file in `main.rs` when deciding which data provider to return from `create_provider()`